lib/f_service/result/base.rb

```diff
@@ -75,7 +75,10 @@ def on(success:, failure:)
       # @return [Success, Failure] the original Result object
       # @api public
       def on_success(*target_types)
-        yield(*to_ary) if successful? && expected_type?(target_types)
+        if successful? && uncaptured? && expected_type?(target_types)
+          yield(*to_ary)
+          @captured = true
```

Comment:
What about freezing the object after this?

Reply:
Done, man: https://github.com/Fretadao/f_service/pull/30/commits/2a2bb4b5b2001362b160df0ea02a17a913e57266